src/persist/src/indexed/mod.rs

```diff
@@ -829,35 +829,54 @@ impl<L: Log, B: Blob> Indexed<L, B> {
         }
 
         for (id, updates) in updates {
-            if let Some(listeners) = self.listeners.get(&id) {
+            if let Some(listeners) = self.listeners.get_mut(&id) {
                 if listeners.is_empty() {
                     continue;
                 }
 
+                let mut dropped_listeners = vec![];
                 let updates = updates
                     .iter()
                     .flat_map(|u| u.iter())
                     .map(|((k, v), ts, diff)| ((k.to_vec(), v.to_vec()), ts, diff))
                     .collect();
 
-                // TODO: remove the listener if it hangs up.
                 if listeners.len() == 1 {
                     let events = ListenEvent::Records(updates);
-                    if let Err(crossbeam_channel::SendError(_)) = listeners[0].send(events) {}
+                    if let Err(crossbeam_channel::SendError(_)) = listeners[0].send(events) {
```

Comment:
let's put a comment somewhere in here that mentions a closes channel means the other end is no longer interested and we can reregister. also add this whole contract to the rustdoc for StreamReadHandle::listen